lib/galaxy/datatypes/text.py

```diff
@@ -48,12 +48,12 @@ def _looks_like_json( self, filename ):
         else:
             with open(filename, "r") as fh:
                 while True:
-                    line = fh.readline()
-                    line = line.strip()
-                    if line:
+                    # Grab first chunk of file and see if it looks like json.
+                    start = fh.read(100).strip()
+                    if start:
                         # simple types are valid JSON as well - but would such a file
                         # be interesting as JSON in Galaxy?
-                        return line.startswith("[") or line.startswith("{")
+                        return start.startswith("[") or start.startswith("{")
```

Comment:
Well, that'd require a bit more code to handle the case where it's a large json file, and split over multiple lines. You'd have to strip whitespace from the file which is starting to sound less "worth it"